app/upload/views.py

```diff
@@ -13,7 +14,15 @@ def upload_document(service_id):
     if 'document' not in request.files:
         return jsonify(error='No document upload'), 400
 
-    document = document_store.put(service_id, request.files['document'])
+    try:
+        virus_free = antivirus_client.scan(request.files['document'])
+    except AntivirusError:
+        return jsonify(error='Antivirus API error'), 503
+
+    if not virus_free:
+        return jsonify(error="Document didn't pass the virus scan"), 400
```

Comment:
It would be nice to return the virus we detected?

Reply:
Antivirus doesn't at the moment. We can add it later, but it's probably worth talking about the security implications of this.